sgv2-docsapi/src/main/java/io/stargate/sgv2/docsapi/api/v2/namespaces/collections/documents/DocumentDeleteResource.java

```diff
@@ -0,0 +1,179 @@
+      @QueryParam("profile") boolean profile) {
+    ExecutionContext context = ExecutionContext.create(profile);
+    return documentWriteService
+        .deleteDocument(namespace, collection, documentId, Collections.emptyList(), context)
+        .onItem()
+        .transform(result -> RestResponse.ResponseBuilder.noContent().build());
+  }
+
+  @Operation(description = "Delete the data at a path in a document by ID.")
+  @Parameters(
+      value = {
+        @Parameter(
+            name = "namespace",
+            ref = OpenApiConstants.Parameters.NAMESPACE,
+            description = "The namespace to write the document to."),
+        @Parameter(
+            name = "collection",
+            ref = OpenApiConstants.Parameters.COLLECTION,
+            description = "The collection to write the document to."),
+        @Parameter(
+            name = "document-id",
+            ref = OpenApiConstants.Parameters.DOCUMENT_ID,
+            description = "The ID of the document that you'd like to write"),
+        @Parameter(
+            name = "document-path",
+            ref = OpenApiConstants.Parameters.DOCUMENT_PATH,
+            description = "The path within the document you would like to write to"),
+        @Parameter(
+            name = "profile",
+            ref = OpenApiConstants.Parameters.PROFILE,
+            description = "Include profiling information from execution."),
+      })
+  @APIResponses(
+      value = {
+        @APIResponse(
+            responseCode = "204",
+            description =
+                "Document updated or created with the provided document-id. The document-id will be returned."),
+        @APIResponse(
+            responseCode = "404",
+            description = "Namespace or collection not found.",
+            content =
+                @Content(
+                    examples = {
+                      @ExampleObject(ref = OpenApiConstants.Examples.NAMESPACE_DOES_NOT_EXIST),
+                      @ExampleObject(ref = OpenApiConstants.Examples.COLLECTION_DOES_NOT_EXIST)
+                    },
+                    schema =
+                        @org.eclipse.microprofile.openapi.annotations.media.Schema(
+                            implementation = ApiError.class))),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_400),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_401),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_500),
+        @APIResponse(ref = OpenApiConstants.Responses.GENERAL_503),
+      })
+  @DELETE
+  @Path("{document-path: .*}")
+  public Uni<RestResponse<Object>> deleteSubDocument(
+      @PathParam("namespace") String namespace,
+      @PathParam("collection") String collection,
+      @PathParam("document-id") String documentId,
+      @PathParam("document-path") List<PathSegment> documentPath,
+      @QueryParam("profile") boolean profile) {
+    ExecutionContext context = ExecutionContext.create(profile);
```

Comment:
And this now brings me to the worry I had when reviewing the `WriteDocumentsService`.. The service should not depend on the table manager at all.. Can we do a quick fix here:

1. remove `TableManager` from the `WriteDocumentsService` 
2. methods that need a table for checking the json schema can accept the `Uni<CqlTable>` as the first parameter
3. resources can then 
    * ensure that table exists (write, update, patch?)
    * get the table in case of delete 
    * I am not sure though if patch should create a new table?